azure/scope/dns.go

```diff
@@ -37,6 +37,11 @@ type DNSScopeParams struct {
 
 	AzureClusterIdentity               infrav1.AzureClusterIdentity
```

Comment:
should we make this `AzureClusterIdentity` more specific now that you added a `ManagementClusterAzureIdentity` ? `what is this identity about?` someone could wonder in the future

Reply:
you mean renaming this to `WorkloadClusterAzureIdentity`?
Are you also fine with comment in the code to describe `AzureClusterIdentity` more explicit?